packages/camera/camera/example/ios/RunnerTests/CameraTestUtils.m

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "CameraTestUtils.h"
+#import <OCMock/OCMock.h>
+@import AVFoundation;
+
+FLTCam *FLTCreateFLTCamWithCaptureSessionQueue(dispatch_queue_t captureSessionQueue) {
+  id inputMock = OCMClassMock([AVCaptureDeviceInput class]);
+  OCMStub([inputMock deviceInputWithDevice:[OCMArg any] error:[OCMArg setTo:nil]])
+      .andReturn(inputMock);
+
+  id sessionMock = OCMClassMock([AVCaptureSession class]);
+  OCMStub([sessionMock alloc]).andReturn(sessionMock);
```

Comment:
This is pretty ugly; please make an init that takes a capture session and expose that in the test method, then have the public init delegate to that one.

Reply:
I agree that it's better to inject these dependencies. As we write more unit tests, there are more dependencies to be injected. This can be a good starting point to adopt this pattern.